### PR TITLE
Use url crate to handle uris

### PR DESCRIPTION
**Pull request by A6GibKm**
Fixes https://github.com/bilelmoussaoui/ashpd/issues/69

### CODE REVIEW

**Comment by A6GibKm** [comment]
I wonder if it makes sense to also use for say, the Screenshot struct which represents a url returned from the portal.

**Comment by bilelmoussaoui** [comment]
> I wonder if it makes sense to also use for say, the Screenshot struct which represents a url returned from the portal.

that would be great. Same for the image uri returned by the account portal

**Comment by bilelmoussaoui** [comment]
the same would be needed for the FileChooser portal as it returns a list of URIs

File: src/desktop/icon.rs [hunk 1]
```diff
@@ -100,8 +103,15 @@ impl TryFrom<OwnedValue> for Icon {
         let type_ = fields[0].downcast_ref::<zvariant::Str>().unwrap();
         match type_.as_str() {
             "file" => {
```
[the remark this answers is not in the document]
Comment (A6GibKm) [reply src/desktop/icon.rs:105]: I don't quite follow

[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply src/desktop/icon.rs:105]: you updated the "file" type, instead of uri?

[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply src/desktop/icon.rs:105]: never mind

**Comment by Be-ing** [comment]
Thanks for implementing this!

**Pull request merged**